docs/literate/src/files/getting_started.jl

```diff
@@ -0,0 +1,288 @@
+#   ````
+#   wget https://julialang-s3.julialang.org/bin/linux/... # your link goes here
+#   ````
+# - Unpack the downloaded file with: 
+#   ````
+#   tar xf julia-....tar.gz # your archive filename goes here
+#   ````
+
+# Now you can verify that Julia is installed entering `<Julia directory>/bin/julia`
+# (e.g. `julia-1.8.5/bin/julia`) command in a terminal. `<Julia directory>` is the directory where
+# Julia is installed.
+# Then Julia will be invoked. To close Julia, enter `exit()` or press `Ctrl+d`.
+
+# Note, that further in the tutorial Julia will be invoked only typing `julia` in a terminal.
+# To enable that, you have to add
+# [Julia to the PATH](https://julialang.org/downloads/platform/#linux_and_freebsd).
+
+# ## Trixi.jl installation
+
+# Trixi.jl and its related tools are registered Julia packages, thus their installation
+# happens inside Julia.
+# For a smooth workflow experience with Trixi.jl, you need to install 
+# [Trixi.jl](https://github.com/trixi-framework/Trixi.jl),
+# [OrdinaryDiffEq.jl](https://github.com/SciML/OrdinaryDiffEq.jl) and 
+# [Plots.jl](https://github.com/JuliaPlots/Plots.jl).
+
+# - Open a terminal and invoke Julia (Windows and Linux: type `julia`).
+# - Execute following commands:
+#   ````
+#   import Pkg
+#   Pkg.add(["OrdinaryDiffEq", "Plots", "Trixi"])
+#   ````
+
+# Now you have installed all these 
+# packages. [OrdinaryDiffEq.jl](https://github.com/SciML/OrdinaryDiffEq.jl) provides time integration schemes
+# used by Trixi.jl and [Plots.jl](https://github.com/JuliaPlots/Plots.jl) can be used to directly
+# visualize Trixi.jl results from the Julia REPL.
+
+# ## Usage
+
+# ### Running a simulation
+
+# To get you started, Trixi.jl has a large set
+# of [example setups](https://github.com/trixi-framework/Trixi.jl/tree/main/examples), that can be taken
+# as a basis for your future investigations.
+# In Trixi.jl, we call these setup files "elixirs", since they contain Julia code that
+# takes parts of Trixi.jl and combines them into something new.
+
+# Now execute one of the examples using the [`trixi_include`](@ref)
+# function. `trixi_include(...)` expects
+# a single string argument with a path to a file containing Julia code.
+# `joinpath(...)` join a path components into a full path. 
+# The [`examples_dir`](@ref) function returns a path to the
+# [`examples`](https://github.com/trixi-framework/Trixi.jl/tree/main/examples) folder
+# that has been locally downloaded while installing Trixi.jl.
+
+# Let's execute a short two-dimensional problem setup. It approximates the solution of
+# the compressible Euler equations in 2D for an ideal gas ([`CompressibleEulerEquations2D`](@ref))
+# with a weak blast wave as the initial condition.
+
+# Invoke Julia in a terminal.
+# And execute following code.
+
+using Trixi, OrdinaryDiffEq
+trixi_include(@__MODULE__,joinpath(examples_dir(), "tree_2d_dgsem", "elixir_euler_ec.jl"))
```

Comment:
Following code runs correctly from julia text file.
`using Trixi, Plots 
trixi_include(joinpath(examples_dir(), "tree_2d_dgsem", "elixir_euler_ec.jl"))
plot(sol)
`
 But when expanding markdown templates during documentation building, `sol` is encountered as undefined:
`[ Info: ExpandTemplates: expanding markdown templates.
┌ Error: failed to run `@example` block in src\tutorials\getting_started.md:120-123
│ ```@example getting_started
│ using Plots
│ plot(sol)
│ ```
│   value = UndefVarError: sol not defined
└ @ Documenter.Expanders C:\Users\Lenovo\.julia\packages\Documenter\H5y27\src\Utilities\Utilities.jl:32`

Reply:
Hm. Maybe you can create [an issue with Literate.jl](https://github.com/fredrikekre/Literate.jl/issues) and ask if this is known behavior or if there is a workaround?

Reply:
If instead of `trixi_include` I call `include`, this issue doesn't occur:
```
using Trixi, Plots 
include(joinpath(examples_dir(), "tree_2d_dgsem", "elixir_euler_ec.jl"))
plot(sol)
```
As I understand it, `trixi_include` in this case does exactly the same, calls `include` only with adding default `Main` module:
```
function trixi_include(mod::Module, elixir::AbstractString; kwargs...)
  # Print information on potential wait time only in non-parallel case
  if !mpi_isparallel()
    @info "You just called `trixi_include`. Julia may now compile the code, please be patient."
  end
  Base.include(ex -> replace_assignments(insert_maxiters(ex); kwargs...), mod, elixir)
end

trixi_include(elixir::AbstractString; kwargs...) = trixi_include(Main, elixir; kwargs...)
```

[The issue with Literate.jl](https://github.com/fredrikekre/Literate.jl/issues/213) is created.

Reply:
Literate.jl developer [answered](https://github.com/fredrikekre/Literate.jl/issues/213#issuecomment-1529736341) on my questions.
Copy of answer:
1) Is it known behavior?
- It is known in the sense that code is not evaluated in Main (by design) but rather in a separate module.
2) Is there way around adding @MODULE?
- Not from the LIterate side. The fact that include_trixi defaults to main is either a feature or design flaw that you would have to discuss with Trixi developers. (A more intuitive behavior, IMO, would be to use a macro and default to the callers current module instead of Main, for example.)

Reply:
Thanks for creating that issue at Literate.jl. Is the following possible:
* **Execute** the `trixi_include(@__MODULE__, ...)` code in the tutorial but **hide** it in the Markdown file
* **Show** the `trixi_include(...)` code in the tutorial (i.e., *without* the `@__MODULE__`) but **do not execute** it?

Could this be a viable workaround?

Reply:
Yes, that's how it should work. With something like this
```
#md # ```julia
#md # using Trixi, Plots
#md # trixi_include(joinpath(examples_dir(), "tree_2d_dgsem", "elixir_euler_ec.jl"))
#md # ```

#md using Trixi, Plots #hide
#md trixi_include(@__MODULE__, joinpath(examples_dir(), "tree_2d_dgsem", "elixir_euler_ec.jl")) #hide

#!md using Trixi, Plots
#!md trixi_include(joinpath(examples_dir(), "tree_2d_dgsem", "elixir_euler_ec.jl"))

# Plot
plot(sol)
```